p2p/libp2p/directSender.go

```diff
@@ -117,7 +117,7 @@ func (ds *directSender) processReceivedDirectMessage(message *pubsubPb.Message,
 	if len(message.TopicIDs) == 0 {
 		return p2p.ErrEmptyTopicList
 	}
-	if ds.checkAndSetSeenMessage(message) {
+	if ds.checkAndSetSeenMessage(message, fromConnectedPeer) {
```

Comment:
Thanks for pointing this out :+1:
I would do it a little bit different though: I would make the node reject the message if the msg.GetFrom() is different from fromConnectedPeer value. So, I would revert the `checkAndSetSeenMessage` function and I would add the following condition on L114:
```GO
	if !bytes.Equal(message.GetFrom(), []byte(fromConnectedPeer)){
		return fmt.Errorf("%w mismatch between From and fromConnectedPeer values", p2p.ErrInvalidMessage)		
	}
```
In this way we would have a stricter way to check the messages.

Reply:
`fromConnectedPeer` may be either local ip or public ip, depending on whether two nodes are in the same net, so such strict condition may cause false negative?

Reply:
libp2p split the peer info into 2: peer ID and multi address data. For example, suppose that we have a connection info such as `/ip4/192.168.0.1/tcp/9999/p2p/16Uiu2HAkw5SNNtSvH1zJiQ6Gc3WoGNSxiyNueRKe6fuAuh57G3Bk`. The string `16Uiu2HAkw5SNNtSvH1zJiQ6Gc3WoGNSxiyNueRKe6fuAuh57G3Bk` is the base58 encoded peer id and the `/ip4/192.168.0.1/tcp/9999` is one entry in the multi-address slice as there can be more entries there (`/ip4/127.0.0.1/tcp/9999`, `/ip4/5.2.221.243/tcp/9999` and so on). So that peer.ID value will always contain an unique string that will have to identify the peer regardless the connection it has to that peer.

Reply:
Good to know! I've updated accordingly.